mlflow/store/file_store.py

```diff
@@ -407,12 +407,14 @@ def _get_run_files(self, run_uuid, resource_type):
     @staticmethod
     def _get_metric_from_file(parent_path, metric_name):
         _validate_metric_name(metric_name)
-        metric_data = read_file_lines(parent_path, metric_name)
+        metric_data = []
+        for line in read_file_lines(parent_path, metric_name):
+            metric_timestamp, metric_value = line.split()
+            metric_data.append((int(metric_timestamp), float(metric_value)))
         if len(metric_data) == 0:
-            raise Exception("Metric '%s' is malformed. No data found." % metric_name)
-        last_line = metric_data[-1]
-        timestamp, val = last_line.strip().split(" ")
-        return Metric(metric_name, float(val), int(timestamp))
+            raise ValueError("Metric '%s' is malformed. No data found." % metric_name)
+        max_timestamp, max_value = sorted(metric_data)[-1]
```

Comment:
This (using `sorted` in this way) is pretty cool, it'd be nice to have a test verifying that the results are sorted by timestamp first & value second, and that append order to the file is ignored(i.e. for `[Metric(key="a", value=2, timestamp=0), Metric(key="a", value=4, timestamp=0), Metric(key="a", value=-1, timestamp=1), Metric(key="a", value=6, timestamp=0)]` we should return `Metric("a", -1, 1)`, etc)

Reply:
+1 to this.

It also just occured to me that `sorted(values)[-1]` is basically the same as `max(values)` - maybe that's the 'right' solution?

Does it also make sense to apply the sorting logic discussed here to the `get_metric_history` method, so that the 'latest' metric value returned here is consistent with the view provided by `get_metric_history` (see also #990)?

Reply:
@acroz @smurching I've adjusted the test logic so that we append metrics with higher timestamps and values first. We then verify that the maximum value at the maximum timestamp is selected even after appending entries with lower timestamps and values.

I've also added an inline comment referencing documentation about comparison behavior between equal-length tuples.

Reply:
@acroz I'll respond to #990 as well, but I think that implementing the same behavior in `get_metric_history` is reasonable for the time being.  Ultimately, the introduction of user-defined x-coordinate values will change the result ordering defined by both of these APIs.